modules/collect/src/test/java/com/opengamma/collect/timeseries/DenseLocalDateDoubleTimeSeriesTest.java

```diff
@@ -0,0 +1,827 @@
+  public void test_of_singleton_nullDateDisallowed() {
+    LocalDateDoubleTimeSeries.of(null, 1d);
+  }
+
+  //-------------------------------------------------------------------------
+  public void test_of_collectionCollection() {
+    Collection<LocalDate> dates = dates(DATE_2011_01_01, DATE_2012_01_01);
+    Collection<Double> values = values(2d, 3d);
+
+    LocalDateDoubleTimeSeries test = LocalDateDoubleTimeSeries.builder().putAll(dates, values).build();
+    assertEquals(test.isEmpty(), false);
+    assertEquals(test.size(), 2);
+    assertEquals(test.containsDate(DATE_2010_01_01), false);
+    assertEquals(test.containsDate(DATE_2011_01_01), true);
+    assertEquals(test.containsDate(DATE_2012_01_01), true);
+    assertEquals(test.get(DATE_2010_01_01), OptionalDouble.empty());
+    assertEquals(test.get(DATE_2011_01_01), OptionalDouble.of(2d));
+    assertEquals(test.get(DATE_2012_01_01), OptionalDouble.of(3d));
+    assertEquals(test.dates().toArray(), new Object[]{DATE_2011_01_01, DATE_2012_01_01});
+    assertEquals(test.values().toArray(), new double[]{2d, 3d});
+  }
+
+  @Test(expectedExceptions = IllegalArgumentException.class)
+  public void test_of_collectionCollection_dateCollectionNull() {
+    Collection<Double> values = values(2d, 3d);
+
+    LocalDateDoubleTimeSeries.builder().putAll(null, values).build();
+  }
+
+  @Test(expectedExceptions = IllegalArgumentException.class)
+  public void test_of_collectionCollection_valueCollectionNull() {
+    Collection<LocalDate> dates = dates(DATE_2011_01_01, DATE_2012_01_01);
+
+    LocalDateDoubleTimeSeries.builder().putAll(dates, null).build();
+  }
+
+  @Test(expectedExceptions = IllegalArgumentException.class)
+  public void test_of_collectionCollection_dateCollectionWithNull() {
+    Collection<LocalDate> dates = Arrays.asList(DATE_2011_01_01, null);
+    Collection<Double> values = values(2d, 3d);
+
+    LocalDateDoubleTimeSeries.builder().putAll(dates, values).build();
+  }
+
+  @Test(expectedExceptions = IllegalArgumentException.class)
+  public void test_of_collectionCollection_valueCollectionWithNull() {
+    Collection<LocalDate> dates = dates(DATE_2011_01_01, DATE_2012_01_01);
+    Collection<Double> values = Arrays.asList(2d, null);
+
+    LocalDateDoubleTimeSeries.builder().putAll(dates, values).build();
+  }
+
+  @Test(expectedExceptions = IllegalArgumentException.class)
+  public void test_of_collectionCollection_collectionsOfDifferentSize() {
+    Collection<LocalDate> dates = dates(DATE_2011_01_01);
+    Collection<Double> values = values(2d, 3d);
+
+    LocalDateDoubleTimeSeries.builder().putAll(dates, values).build();
+  }
+
+  public void test_of_collectionCollection_datesUnordered() {
+    Collection<LocalDate> dates = dates(DATE_2012_01_01, DATE_2011_01_01);
+    Collection<Double> values = values(2d, 1d);
+
+    LocalDateDoubleTimeSeries series = LocalDateDoubleTimeSeries.builder().putAll(dates, values).build();
```

Comment:
It may be that the `LocalDateDoubleTimeSeries.of(Collection, Collection)` is needed for test cases after all. Or it may be that those test cases would be better using the builder properly. Not sure.